torch/_torch_docs.py

```diff
@@ -3508,25 +3508,164 @@ def merge_dicts(*dicts):
+when the multiplication :math:`h_r h_l \rightarrow 0` then we have:
+
+.. math::
+    f'(x) =  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+             + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+             - \mathcal{O}({h_r}{h_l}).
+
+
+Therefore, we can estimate :math:`f'(x)` by
+
+.. math::
+    f'(x) \approx  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+          + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+
+An interesting special case, happens when :math:`h_r = h_l=h` which gives us
+:math:`f'(x) \approx  \frac{  f(x+h) - f(x-h) }{ 2h}`.
+
+.. note::
+    We estimate the gradient of functions in complex domain
+    :math:`g : \mathbb{C}^n \rightarrow \mathbb{C}` in the same way.
 
 Args:
-    {input}
+    input (``Tensor``): the tensor that represents the values of the function
 
 Keyword args:
-    spacing (scalar, list of scalar, list of Tensor, optional): implicitly or explicitly represents
-    the coordinates the function is evaluated at
-    dim (int, list of int, optional): the dimension or dimensions to approximate the gradient over.
-    edge_order (int, optional): unsupported (must be equal to its default value which is 1.)
+    spacing (``scalar``, ``list of scalar``, ``list of Tensor``, optional): i) list of 1-dimensional Tensors
+        where each of them would represent explicit coordinates in the corresponding
+        directions.
+        ii) list of scalars where each scalar implicitly represents coordinates as the step size
+        in the corresponding direction as we discussed above.
+        iii) scalar, this is also an implicit representation of coordinates where step size in all
+        directions is the same.
+        iv) not provided, this case is equivalent to implicit representation where all coordinates
+        have a step size of 1.
 
-Example:
+    dim (``int``, ``list of int``, optional): the dimension or dimensions to approximate the gradient over.
+        When this parameter has not been provided, then we compute gradients in all dimensions.
+
+    edge_order (``int``, optional): 1 or 2, depending on whether we want to have a second-order or
+        first-order estimation for edge points.
 
-    >>> t = torch.tensor([1, 2, 4, 7, 11, 16], dtype=torch.float)
-    >>> torch.gradient(t)
-    tensor([1. , 1.5, 2.5, 3.5, 4.5, 5. ])
-    >>> coords = torch.tensor([0., 1., 1.5, 3.5, 4., 6.], dtype=torch.float)
-    >>> torch.gradient(t, spacing=(coords,))
-    tensor([1. ,  3. ,  3.5,  6.7,  6.9,  2.5])
+.. note::
+    When the ``dim`` argument is available, then the ``spacing`` argument is presented in the same order
+    with respect to dim. However, when ``dim`` is not provided, then the ``spacing`` argument can only be
+    a scalar (also optional) or if it is given as a list of scalar/Tensor with the size of the list to be
+    equal to n.
+
+Example::
+
+    >>> t = torch.tensor([[1, 2, 4, 8], [10, 20, 40, 80]])
```

Comment:
Is there an interesting function to estimate the gradient of?

Reply:
I agree with that starting with a nice function and giving estimates of its gradient is an amazing idea.

However, i believe in this case, we want to illustrate the effect of spacing/coordinates in the values of gradient estimation. In order to do this illustration to fix an output data and playing with spacing argument in showing them they are giving different output is the path we chose here for this purpose.

Reply:
I agree the current examples do a good job demonstrating how spacing works. An additional example with an interesting function that demonstrates the quality of the approximation might be interesting, too

Reply:
after the discussion, we decided to add f(x)=x^2 as a simple illustrative function of the operator.